### PR TITLE
#11 setup of required data

### PR DESCRIPTION
**Pull request by mirkopleli**

### CODE REVIEW

File: src/services/config.service.js [hunk 1]
```diff
@@ -13,6 +13,15 @@ class ConfigService {
       })
       .catch(e => console.log('No master password found'))
   }
+
+  async setMasterPassword (masterPassword) {
+    return $db().collection(COLLECTION_NAME).doc('password').set({
+      value: masterPassword
+    })
+      .catch(function (error) {
```
Comment (stritti) [inline src/services/config.service.js:21]: Remove 'funktion' keyword.
On error user should be notified, what happens, if no master password is set on the other hand?

File: src/views/setup/Building.vue [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<template>
+  <b-container class="mt-3 align-content-center">
+    <b-row>
+      <h3 class="text-center pb-3">Welcome to the Setup from the COVID-Bouncer Admin</h3>
```
Comment (stritti) [inline src/views/setup/Building.vue]: Welcome to the setup of COVID-Bouncer

File: src/views/setup/Location.vue [hunk 3]
```diff
@@ -0,0 +1,88 @@
+<template>
+  <b-container class="mt-3 align-content-center">
+    <b-row>
+      <h3 class="text-center pb-3">Welcome to the Setup from the COVID-Bouncer Admin</h3>
```
Comment (stritti) [inline src/views/setup/Location.vue]: Unify titles as proposed above

File: src/views/setup/Location.vue [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  }
+}
+</script>
+
+<style lang="sass" scoped>
+.nextButton
+  border: none
+  font-weight: bold
+
+.input-styling:focus
```
Comment (stritti) [inline src/views/setup/Location.vue]: Define global for all controls

File: src/views/setup/Location.vue [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        title: 'Important message',
+        text: message
+      })
+    }
+  }
+}
+</script>
+
+<style lang="sass" scoped>
+.nextButton
```
Comment (stritti) [inline src/views/setup/Location.vue]: Define global for all

File: src/views/setup/MasterPassword.vue [hunk 6]
```diff
@@ -0,0 +1,66 @@
+<template>
+<b-container class="mt-3 align-content-center">
+    <b-row>
+      <h3 class="text-center pb-3">Welcome to the Setup from the COVID-Bouncer Admin</h3>
```
Comment (stritti) [inline src/views/setup/MasterPassword.vue]: See above

File: src/views/setup/Building.vue (hunk 2 above)
Comment (chrisWolter) [inline src/views/setup/Building.vue]: Typo: Setup **of** the Covid Bouncer

File: src/views/setup/Building.vue [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+          label="3. Step: Add a Building"
+          id="building"
+          class="input-styling">
+          <b-form-input
+            id="input-building"
+            v-model="building"
+            :state="invalidFeedback"
+            trim>
+          </b-form-input>
+          <b-form-text id="initials-help-block">
```
Comment (chrisWolter) [inline src/views/setup/Building.vue:18]: Why cant the "add building" step be skipped? As a User i do not have to add a building.

File: src/views/setup/Building.vue [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            This step can not be skipped
+          </b-form-text>
+          <!-- This will only be shown if the preceding input has an invalid state -->
+          <b-form-invalid-feedback id="input-live-feedback">
```
Comment (chrisWolter) [inline src/views/setup/Building.vue:22]: This block is obsolet.

File: src/views/setup/Building.vue [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    }
+  },
+  methods: {
+    addBuilding: async function () {
+      try {
+        await locationService.addBuilding(this.building, this.location)
+      } catch (error) {
+        this.showErrorNotification('Error while adding a new building')
+      }
+      await this.$router.push({ name: 'Home' })
```
Comment (chrisWolter) [inline src/views/setup/Building.vue:54]: remove await

File: src/views/setup/Location.vue (hunk 3 above)
Comment (chrisWolter) [inline src/views/setup/Location.vue]: Typo

File: src/views/setup/Location.vue [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+          label="2. Step: Add a location"
+          id="location"
+          class="input-styling">
+          <b-form-input
+            id="input-location"
+            v-model="location"
+            :state="invalidFeedback"
+            trim>
+          </b-form-input>
+          <b-form-text id="initials-help-block">
```
Comment (chrisWolter) [inline src/views/setup/Location.vue:18]: Why cant this step be skipped or is this block obsolet?

File: src/views/setup/Location.vue [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+            This step can be skipped
+          </b-form-text>
+          <!-- This will only be shown if the preceding input has an invalid state -->
+          <b-form-invalid-feedback id="input-live-feedback">
```
Comment (chrisWolter) [inline src/views/setup/Location.vue:22]: obsolet block

File: src/views/setup/Location.vue [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+  },
+  methods: {
+    addLocation: async function () {
+      try {
+        const response = await locationService.addLocation(this.location)
+        this.locationId = response.id
+      } catch (error) {
+        this.showErrorNotification('Error while adding a new Location.')
+      }
+      await this.$router.push({ name: 'Building', params: { location: this.locationId } })
```
Comment (chrisWolter) [inline src/views/setup/Location.vue:63]: remove await

File: src/views/setup/MasterPassword.vue (hunk 6 above)
Comment (chrisWolter) [inline src/views/setup/MasterPassword.vue]: Typo: from --> of

File: src/views/setup/MasterPassword.vue [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    </b-row>
+  <b-row>
+    <form>
+        <b-form-group
+          label="1. Step: Setup the master password"
+          id="master-password"
+          class="input-styling">
+          <b-form-input
```
Comment (chrisWolter) [inline src/views/setup/MasterPassword.vue:12]: There should be a check for minimal password length. I would suggest 8 chars.

File: src/views/setup/MasterPassword.vue [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+  </b-row>
+</b-container>
+</template>
+
+<script>
+import { configService } from '@/services/config.service'
+export default {
+  computed: {
+    invalidFeedback () {
+      return this.masterPassword.length > 5
```
Comment (chrisWolter) [inline src/views/setup/MasterPassword.vue]: set password length to 8 for security reasons.

File: src/views/setup/MasterPassword.vue [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+            v-model="masterPassword"
+            :state="invalidFeedback"
+            trim>
+          </b-form-input>
+          <b-form-text id="initials-help-block">
+            This step can not be skipped
+          </b-form-text>
+          <!-- This will only be shown if the preceding input has an invalid state -->
+          <b-form-invalid-feedback id="input-live-feedback">
+            Enter at least 5 letters
```
Comment (chrisWolter) [inline src/views/setup/MasterPassword.vue]: 5 --> 8

File: src/views/setup/MasterPassword.vue [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+    }
+  },
+  methods: {
+    setMasterPassword: async function () {
+      try {
+        await configService.setMasterPassword(this.masterPassword)
+      } catch (error) {
+        this.showErrorNotification('Error while setting a new master password')
+      }
+      await this.$router.push({ name: 'Location' })
```
Comment (chrisWolter) [inline src/views/setup/MasterPassword.vue:52]: please remove await

**Review by chrisWolter — CHANGES_REQUESTED** [review]
- [x] It should be possible to skip the add building step
- [x] The invalid input error should only show up on submit and not on start.

**Review by stritti — CHANGES_REQUESTED** [review]

File: src/views/setup/Building.vue (hunk 9 above)
Comment (mirkopleli) [reply src/views/setup/Building.vue:54]: image

File: src/views/setup/MasterPassword.vue (hunk 13 above)
Comment (mirkopleli) [reply src/views/setup/MasterPassword.vue:12]: True

File: src/services/config.service.js (hunk 1 above)
Comment (mirkopleli) [reply src/services/config.service.js:21]: This line of code is from Firebase itself, at this point. That here an error comes is not possible except our Firestor has a problem. The password is simply saved in the service.

**Review by stritti — APPROVED** [review]

**Review by chrisWolter — APPROVED** [review]

**Pull request merged**